model/src/main/kotlin/licenses/LicenseConfiguration.kt

```diff
@@ -51,6 +51,32 @@ data class LicenseConfiguration(
             }
         }
     }
+
+    /**
+     * A property for fast look-ups of licenses for a given set.
+     */
+    private val licensesBySetId by lazy {
```

Comment:
You should define the return type as `Map<String, Set<License>>` here to avoid returning mutable sets.